### PR TITLE
Conditional disable android buildin extensions

### PR DESCRIPTION
**Pull request by heke123**
[Crosswalk-Lite][Android][Feature XWALK-3436]

Disable/Enable is controlled by flag:"disable_android_buildin_extensions=1/0".

By default buildin extensions are Disabled in Xwalk-Lite.

### CODE REVIEW

**Comment by heke123** [comment]
@halton 
Please help give a review on feature [conditional disable android buildin extensions,  XWALK-3436]

**Comment by halton** [comment]
Please follow the git log format, suggest as 
```
[Lite] Conditional disable Android buildin extensions 

Disable/Enable is controlled by the GYP flag disable_android_buildin_extensions. This flag is 1 for the Lite branch.

BUG=XWALK-3436
```

File: build/android/generate_xwalk_core_library.py [hunk 1]
```diff
@@ -81,12 +81,9 @@ def CopyJSBindingFiles(project_source, out_dir):
...
       'xwalk/experimental/presentation/presentation_api.js',
-      'xwalk/runtime/android/core_internal/src/org/xwalk/core/'
-      + 'internal/extension/api/contacts/contacts_api.js',
-      'xwalk/runtime/android/core_internal/src/org/xwalk/core/'
-      + 'internal/extension/api/device_capabilities/device_capabilities_api.js',
-      'xwalk/runtime/android/core_internal/src/org/xwalk/core/'
-      + 'internal/extension/api/messaging/messaging_api.js'
+      'xwalk/runtime/android/core_internal/extension/api/contacts/contacts_api.js',
+      'xwalk/runtime/android/core_internal/extension/api/device_capabilities/device_capabilities_api.js',
+      'xwalk/runtime/android/core_internal/extension/api/messaging/messaging_api.js'
```
Comment (halton) [inline build/android/generate_xwalk_core_library.py]: I think this will not pass the PEP8 line limit(80 chars)

File: build/common.gypi [hunk 2]
```diff
@@ -4,12 +4,14 @@
     'tizen_mobile%': 0,
     'shared_process_mode%': 0,
     'enable_murphy%': 0,
+    'disable_android_buildin_extensions%': 1,
```
Comment (halton) [inline build/common.gypi]: Set default as 0 please, this PR suppose back to master. We should control it in ./xwalk/gyp_xwalk

File: runtime/android/core_internal/extension/README.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+java files in this folder is moved from .../runtime/android/core_internal/src/.../internal/extension.
+The build-in extensions can be conditionally set to disable/enable before compiling by using flag "disable_android_buildin_extensions = 0/1". 
+so the java files in this folder won't be compiled if the build-in extensions are disabled.
```
Comment (halton) [inline runtime/android/core_internal/extension/README.txt]: I do not think this readme is needed.

File: test/android/runtime_client/javatests/extension_tests/README.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+java files in this folder is moved from test/runtime_client/src/org/.../test.
+test cases in this folder have dependency on the build-in extensions which could be conditionally disabled. 
+so the java files in this folder won't be compiled if the build-in extensions are disabled.
```
Comment (halton) [inline test/android/runtime_client/javatests/extension_tests/README.txt]: ditto

File: test/android/runtime_client_embedded/javatests/extension_tests/README.txt [hunk 5]
```diff
@@ -0,0 +1,3 @@
+java files in this folder is moved from test/runtime_client/src/org/.../test.
+test cases in this folder have dependency on the build-in extensions which could be conditionally disabled. 
+so the java files in this folder won't be compiled if the build-in extensions are disabled.
```
Comment (halton) [inline test/android/runtime_client_embedded/javatests/extension_tests/README.txt]: ditto

File: build/android/generate_xwalk_core_library.py (hunk 1 above)
Comment (halton) [inline build/android/generate_xwalk_core_library.py]: And how about launch_screen and presentation apis? I think all those js files should be under protection of condition that `disable_android_buildin_extensions`

File: xwalk_android.gypi [hunk 6]
```diff
@@ -276,9 +283,9 @@
           'files': [
             'experimental/launch_screen/launch_screen_api.js',
             'experimental/presentation/presentation_api.js',
-            'runtime/android/core_internal/src/org/xwalk/core/internal/extension/api/contacts/contacts_api.js',
-            'runtime/android/core_internal/src/org/xwalk/core/internal/extension/api/device_capabilities/device_capabilities_api.js',
-            'runtime/android/core_internal/src/org/xwalk/core/internal/extension/api/messaging/messaging_api.js',
+            'runtime/android/core_internal/extension/api/contacts/contacts_api.js',
+            'runtime/android/core_internal/extension/api/device_capabilities/device_capabilities_api.js',
+            'runtime/android/core_internal/extension/api/messaging/messaging_api.js',
```
Comment (halton) [inline xwalk_android.gypi]: ditto

File: xwalk_android_tests.gypi [hunk 7]
```diff
@@ -80,9 +80,9 @@
           'files': [
             'experimental/launch_screen/launch_screen_api.js',
             'experimental/presentation/presentation_api.js',
-            'runtime/android/core_internal/src/org/xwalk/core/internal/extension/api/contacts/contacts_api.js',
-            'runtime/android/core_internal/src/org/xwalk/core/internal/extension/api/device_capabilities/device_capabilities_api.js',
-            'runtime/android/core_internal/src/org/xwalk/core/internal/extension/api/messaging/messaging_api.js',
+            'runtime/android/core_internal/extension/api/contacts/contacts_api.js',
+            'runtime/android/core_internal/extension/api/device_capabilities/device_capabilities_api.js',
+            'runtime/android/core_internal/extension/api/messaging/messaging_api.js',
```
Comment (halton) [inline xwalk_android_tests.gypi]: ditto

File: xwalk_android_tests.gypi [hunk 8]
```diff
@@ -360,9 +360,9 @@
...
```
Comment (halton) [inline xwalk_android_tests.gypi]: ditto

**Comment by halton** [comment]
Close this PR since #2844 is new for it. 
We've changed the branch name, so this close-recreate one does make sense.

**Pull request closed without merging**